scene/gui/rich_text_label.cpp

```diff
@@ -3367,14 +3380,46 @@ Error RichTextLabel::append_bbcode(const String &p_bbcode) {
 	return OK;
 }
 
+void RichTextLabel::scroll_to_paragraph(int p_paragraph) {
+	ERR_FAIL_INDEX(p_paragraph, main->lines.size());
+	_validate_line_caches(main);
+	vscroll->set_value(main->lines[p_paragraph].offset.y);
+}
+
+int RichTextLabel::get_paragraph_count() const {
+	return current_frame->lines.size();
+}
+
+int RichTextLabel::get_visible_paragraph_count() const {
+	if (!is_visible()) {
+		return 0;
+	}
+	return visible_paragraph_count;
+}
+
 void RichTextLabel::scroll_to_line(int p_line) {
-	ERR_FAIL_INDEX(p_line, main->lines.size());
 	_validate_line_caches(main);
-	vscroll->set_value(main->lines[p_line].offset.y);
+
+	int line_count = 0;
+	for (int i = 0; i < main->lines.size(); i++) {
+		if ((line_count <= p_line) && (line_count + main->lines[i].text_buf->get_line_count() >= p_line)) {
+			float line_offset = 0.f;
+			for (int j = 0; j < p_line - line_count; j++) {
+				line_offset += main->lines[i].text_buf->get_line_size(j).y;
+			}
+			vscroll->set_value(main->lines[i].offset.y + line_offset);
+			return;
+		}
+		line_count += main->lines[i].text_buf->get_line_count();
+	}
 }
 
 int RichTextLabel::get_line_count() const {
-	return current_frame->lines.size();
+	int line_count = 0;
+	for (int i = 0; i < main->lines.size(); i++) {
+		line_count += main->lines[i].text_buf->get_line_count();
+	}
+	return line_count;
 }
```

Comment:
I think this is wrong, or should at least be made optional.

The actual amount of lines in a RTL and the amount of lines _rendered_ is two different things.

When you want to get the line count of the number of lines the actual RLT contains, for example to remove a line in `remove_line()`, you want the value of `current_frame->lines.size()`, not including wrapped lines. If you include wrapped lines, you may be reporting a larger number of lines than actually exists. In which case you will exit out of `remove_line()` early, since `p_line > current_frame->lines.size()`

Reply:
`get_paragraph_count` should return number of lines without counting wrapped lines.

Reply:
I see... sorry, didn't catch that.

Were uses of the "legacy" `get_line_count()` changed to `get_paragraph_count()` within the rest of the codebase?

I'm also not sure about the naming... but I guess it works and I don't have better suggestions at this stage.